design/proposals/encoded_token_response.md

```diff
@@ -0,0 +1,148 @@
+| JWT      |json web token    |
+
+
+## Useful links
+1. [HTTP Content Negotiation Best Practices](https://restfulapi.net/content-negotiation/)
+2. MDN web docs:
+    1. [HTTP Accept Header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept)
+    2. [HTTP Accept-Encoding Header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Encoding)
+    3. [HTTP Content-Type Header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type)
+    4. [HTTP Content-Encoding Header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding)
+3. [Rails Routing Guides](https://guides.rubyonrails.org/routing.html#route-globbing-and-wildcard-segments)
+4. [Correct Usage of JSON](https://stackoverflow.com/questions/477816/what-is-the-correct-json-content-type)
+
+## Background
+
+As a Conjur User or Host, I want to send authentication request to `authn` authenticator (for example).
+So that I receive a JWT access token.
+
+![alt text](encoded_token_chart.jpeg "Login process")
+
+Current authentication response is a Flattened JWS JSON Serialization, e.g:
+```json
+{
+    "protected": "eyJh...",
+    "payload": "eyJz...",
+    "signature": "CZgB..."
+}
+``` 
+
+In order to use this token for restricted endpoint it **must** be encode and added to `Authorization: Token token=` header:
+```sh
+echo "Get Conjur access token using an Azure access token"
+# Get an authn-azure Conjur access token for host azure-apps/test-app
+authn_azure_response=$(curl -k -X POST \
+  -H "Content-Type: application/x-www-form-urlencoded" \
+  --data "jwt=$azure_access_token" \
+  https://"$CONJUR_SERVER_DNS":8443/authn-azure/test/cucumber/host%2Fazure-apps%2Ftest-app/authenticate)
+authn_azure_access_token=$(echo -n "$authn_azure_response" | base64 | tr -d '\r\n')
+
+echo "Retrieve a secret using the Conjur access token"
+# Retrieve a Conjur secret using the authn-azure Conjur access token
+secret=$(curl -k -H "Authorization: Token token=\"$authn_azure_access_token\"" \
+  https://"$CONJUR_SERVER_DNS":8443/secrets/cucumber/variable/secrets/test-variable)
+
+echo "Retrieved secret ${secret} from Conjur!!!"
+```
+[Code reference](https://github.com/cyberark/conjur/blob/application_identity_validation/ci/authn-azure/run-authn-azure.sh#L16)
+
+
+The requested situation is to support both response of current **Flattened** JWS JSON Serialization with content type `application/json` and encoded JWS **Compact** Serialization with content type `text\plain`. 
+
+## Issue description
+1. HTTP server best practices: Using `application\json` as the content type to encoded output (not a JSON) is bad practice.
+    So encoded output shouldn't use `application/json` unless it will be in JSON format. 
+    The `../autheticate` response is in JSON format so it makes sense to use `application\json` as the response content type.
+        As well as `../logic`,  which returns the api key as `text\plain`.
+
+## Solution - Accept HTTP Header
```

Comment:
Comment from meeting:
If user supply accept: 'xml' he will get json
even if user supply accept 'json' he will get 'json' but it's not because he specify json it is because of default behavior, so we must specify that the only excepted header is "text\plain" in it's full qualifier name ("txt" will not be parsed as "text\plain" thus return json)

Reply:
Added